packages/commerce-sdk-react/src/hooks/ShopperStores/query.ts

```diff
@@ -38,18 +39,19 @@ export const useSearchStores = (
     type Data = DataType<Client['searchStores']>
     const {shopperStores: client} = useCommerceApi()
     const methodName = 'searchStores'
-    const requiredParameters = ['organizationId', 'siteId'] as const
+    const requiredParameters = ShopperStores.paramKeys[`${methodName}Required`]
 
     // Parameters can be set in `apiOptions` or `client.clientConfig`;
     // we must merge them in order to generate the correct query key.
     const netOptions = omitNullableParameters(mergeOptions(client, apiOptions))
+    const parameters = pickValidParams(netOptions.parameters, ShopperStores.paramKeys[methodName])
     const queryKey = queryKeyHelpers[methodName].queryKey(netOptions.parameters)
     // We don't use `netOptions` here because we manipulate the options in `useQuery`.
     const method = async (options: Options) => await client[methodName](options)
 
     // For some reason, if we don't explicitly set these generic parameters, the inferred type for
     // `Data` sometimes, but not always, includes `Response`, which is incorrect. I don't know why.
-    return useQuery<Client, Options, Data>(netOptions, queryOptions, {
+    return useQuery<Client, Options, Data>({...netOptions, parameters}, queryOptions, {
```

Comment:
This change is different than the other ones. Because in this case, we forgot to filter out the invalid parameters, which is what the other files have done.

Reply:
Aside: this is another reason for having templates to generate the code. With it, we can avoid inconsistency/bug like this.